apps/api/v2/src/modules/workflows/inputs/create-workflow.input.ts

```diff
@@ -0,0 +1,229 @@
+import { ApiProperty, ApiPropertyOptional } from "@nestjs/swagger";
+import { Type } from "class-transformer";
+import {
+  IsBoolean,
+  IsEmail,
+  IsEnum,
+  ArrayMinSize,
+  IsNumber,
+  IsOptional,
+  IsString,
+  ValidateNested,
+  ValidateIf,
+} from "class-validator";
+
+export enum WorkflowTriggerType {
+  BEFORE_EVENT = "BEFORE_EVENT",
+  EVENT_CANCELLED = "EVENT_CANCELLED",
+  NEW_EVENT = "NEW_EVENT",
+  AFTER_EVENT = "AFTER_EVENT",
+  RESCHEDULE_EVENT = "RESCHEDULE_EVENT",
+}
+
+export enum WorkflowTimeUnit {
+  HOUR = "HOUR",
+  MINUTE = "MINUTE",
+  DAY = "DAY",
+}
+
+export enum RecipientType {
+  HOST = "HOST",
+  ATTENDEE = "ATTENDEE",
+  EMAIL = "EMAIL",
+  PHONE_NUMBER = "PHONE_NUMBER",
+}
+
+export enum StepAction {
+  EMAIL_HOST = "EMAIL_HOST",
+  EMAIL_ATTENDEE = "EMAIL_ATTENDEE",
+  EMAIL_ADDRESS = "EMAIL_ADDRESS",
+  SMS_ATTENDEE = "SMS_ATTENDEE",
+  SMS_NUMBER = "SMS_NUMBER",
+  WHATSAPP_ATTENDEE = "WHATSAPP_ATTENDEE",
+  WHATSAPP_NUMBER = "WHATSAPP_NUMBER",
+}
+
+export enum TemplateType {
+  REMINDER = "REMINDER",
+  CUSTOM = "CUSTOM",
+}
+
+export class WorkflowActivationDto {
+  @ApiProperty({
+    description: "Whether the workflow is active for all calendars",
+    example: false,
+    type: Boolean,
+  })
+  @IsBoolean()
+  isActiveOnAll!: boolean;
+
+  @ApiPropertyOptional({
+    description: "List of active calendar IDs the workflow applies to, required if isActiveOnAll is false",
+    example: [698191],
+    type: [Number],
+  })
+  @ValidateIf((o) => !Boolean(o.isActiveOnAll))
+  @IsOptional()
+  @IsNumber({}, { each: true })
+  activeOnEventTypeIds?: number[] = [];
+}
+
+export class WorkflowTriggerOffsetDto {
+  @ApiProperty({ description: "Time value for offset before/after event trigger", example: 24, type: Number })
+  @IsNumber()
+  value!: number;
+
+  @ApiProperty({ description: "Unit for the offset time", example: "HOUR", enum: WorkflowTimeUnit })
+  @IsEnum(WorkflowTimeUnit)
+  unit!: WorkflowTimeUnit;
+}
+
+export class WorkflowTriggerDto {
+  @ApiProperty({
+    description: "Trigger type for the workflow",
+    example: "BEFORE_EVENT",
+    enum: WorkflowTriggerType,
+  })
+  @IsEnum(WorkflowTriggerType)
+  type!: WorkflowTriggerType;
+
+  @ApiPropertyOptional({
+    description: "Offset before/after the trigger time; required for BEFORE_EVENT and AFTER_EVENT only",
```

Comment:
It's an indication to me that each workflow trigger should have it's own DTO and then we have pipe parsing and validating each of them based on the `type` passed.